build/helper/metadata_add_all.py

```diff
@@ -297,6 +297,15 @@ def _fix_type(parameter):
     parameter['type'] = parameter['type'].replace('[ ]', '[]').replace(' []', '[]').replace(' ', '_')
 
 
+def _fix_custom_type(parameter, config):
+    '''Add "struct_" prefix to custom type if necessary to match its ctypes_type.'''
```

Comment:
This function is "monkey patching" the metadata. It's hacky. We have full control of the metadata.

Do we want to enforce a convention that a custom type and the internal ctypes type must have the same name except for the prefix? Then let's add a check and fix the metadata instead. It's more work but more correct in the long term.

If so, then let's add a strict fix

Reply:
Following our offline discussion, we have agreed to proceed with the current approach as a temporary workaround to unblock the LCR-related PRs. We shall update the codegen logic for handling custom types once the metadata have been fixed from the upstream.